src/elements/TextInput.tsx

```diff
@@ -2,25 +2,32 @@ import {
   FocusEvent,
   ForwardRefRenderFunction,
   ForwardedRef,
+  InputHTMLAttributes,
   forwardRef,
   useState,
 } from "react";
-import InputUnstyled, { InputUnstyledProps } from "@mui/base/InputUnstyled";
+import { useInput } from "@mui/base";
 import { Box, Stack } from "@mui/material";
 import styled from "styled-components";
 import theme from "theme";
 import Typography from "./Typography";
 
-export interface TextInputProps extends Omit<InputUnstyledProps, "helperText"> {
+export interface TextInputProps extends InputHTMLAttributes<HTMLInputElement> {
   readonly label?: string;
   readonly errorMessage?: string;
+  readonly startAdornment?: JSX.Element;
+  readonly endAdornment?: JSX.Element;
 }
 
 const StyledRootElement = styled.div`
   display: flex;
   justify-content: space-between;
   align-items: center;
   flex-grow: 1;
+  padding: ${theme.inputField.padding};
```

Comment:
I noticed that this var is set to `9px 13px` which is what figma has, but it doesn't go well with the 4px grid system and in the long run it will throw things off by a pixel or 2. I was going to suggest we change that value to `8px 12px`. What do you think?

Reply:
Also, do we want to tie the styles of the root element to the `theme.inputField.padding`? 
I see we are adding padding in the root and in the input, we should only be adding it to one right? 
I might be missing something here though 🤔

Reply:
For the pixels, I was just going by the dimensions in the Figma mock-up, but we could change it.

For the padding, unfortunately, the start and end adornments look incorrect if we only add padding to the input, since the padding needs to be outside of them. I'm realizing that I need to do the same thing for the left padding though, so tha the "start" adornment will show up correctly.

Reply:
Let's check with design and understand where they were going with this. I dont think this is something that needs to be fixed with this PR, but worth a discussion.